### PR TITLE
Missing break in trap scheduler func

### PR DESCRIPTION
**Pull request by donnierussellii**
Clearly should be there; wonder what it fixes/breaks

### CODE REVIEW

File: src/GameSrc/trigger.c [hunk 1]
```diff
@@ -648,8 +648,10 @@ errtype trap_scheduler_func(int p1, int p2, int p3, int p4) {
         switch (objs[current_trap].obclass) {
         case CLASS_TRAP:
             p = &(objTraps[objs[current_trap].specID].p3);
+            break; //looks like there should be a break here!
         case CLASS_FIXTURE:
             p = &(objFixtures[objs[current_trap].specID].p3);
+            break; //feeling generous, one here too
```
Comment (Interrupt) [inline src/GameSrc/trigger.c]: This one isn't really needed if it's the last one in the list, nothing to break out of right?

**Comment by Interrupt** [comment]
How much testing did you do around traps after making this change? You're right in wondering how much of the game this could affect :D

File: src/GameSrc/trigger.c (hunk 1 above)
Comment (donnierussellii) [reply src/GameSrc/trigger.c]: I brake for breaks. Going for broke.

**Comment by donnierussellii** [comment]
I only played through level 1 to test it and didn't notice anything peculiar.

**Comment by donnierussellii** [comment]
I used a recharging station, waited for it to recharge then used it again, and that worked.
I also blew up the four antenna relays, and that worked.
I know these use the p3 trap variable, which is counted down in this function.

The trap case gets called continously on most levels; on some not at all. I haven't seen any calls of the fixture case.

But because of the missing break; the trap case was falling through to the fixture case and replacing the pointer, then counting that down.

Since I can't understand what's going on here, I'm closing this.

**Comment by donnierussellii** [comment]
Unless this is ever true (I haven't seen it so):
        if ((p3 < 0x1000) && (p3 > 0))
            (*p)--;
the missing break; is irrelevant.

**Pull request closed without merging**